docs/network_new/README.md

```diff
@@ -0,0 +1,84 @@
+![like so](https://github.com/threefoldtech/zos/blob/master/specs/network/HIDDEN-PUBLIC.png)
+
+- **wireguard port management**  
+Every wireguard point (a network resource point) needs a destination/port combo when it's  publicly reachable. The destination is a public ip, but the port is the differentiator. So we need to make sure every nwetwork wireguard listening port is unique in the node wehere it runs, and can be reapplied in case of a node's reboot.
+ZOS registers the ports **already in use** to the BCDB, so a user can the pick a port that is not yet used.
+
+- **wireguard and hidden nodes**  
+Hidden nodes are nodes that are in essence hidden behind a firewall, and unreachable from the Internet to an internal net, be it as an IPv4 NATed host or an IPv6 host that is firewalled in any way, where it's impossible to have connection initiations form the Internet to the node.  
+As such, these nodes can only partake in a network as client-only towards publicly reachable peers, and can only initiate the connections themselves. (ref previous drawing).  
+To make sure connectivity stays up, the clients (all) have a keepalive towards all their peers so that communications towards network resources in hidden nodes can be established.
+
+## caveats
+
+- **hidden nodes**  
+Hidden nodes live (mostly) behind firewalls that keep state about connections and these states have a lifetime. We try at best to keep these communications going, but depending of the firewall your mileage may vary (YMMV ;-))
+
+- **local underlay network reachability**  
+When multiple nodes live in a same hidden network, at the moment we don't try to have the nodes establish connectivity between themselves, so all nodes in that hidden network can only reach each other through the intermediary of a node that is publicly reachable. So to get some performance, a farmer will have to have real routable nodes available in the vicinity.
+So for now, a farmer is better off to have his nodes really reachable over a public network.
+
+- **IPv6 and IPv4 considerations**  
+While the mesh can work over IPv4 __and__ IPv6 at the same time, the peers can only be reached through one protocol at the same time. That is a peer is IPv4 __or__ IPv6, not both. Hence if a peer is reachable over IPv4, the client towards that peer needs to reach it over IPv4 too and thus needs an IPv4 address.
+We advise strongly to have all nodes properly set-up on a routable unfirewalled IPv6 network, so that these problems have no reason to exist.
+
+## future
+
+- **CNI** 
+ZOS and it's Wireguard mesh per user is a quite novel way to do things, but there are many overlay networks that are built to solve other network requirements in very different ways, and these solutions could be intagrated in a later phase, through the use of `CNI`, a common way to request a network for a user, or for a specific workload.
+
+- **automated provisioning**  
```

Comment:
As far as I know this is not on the roadmap at all. So I would remove this paragraph altogether